parsers/ENTSOE.py

```diff
@@ -1208,61 +1208,69 @@ def get_raw_exchange(
+        for domain_pair in EXCHANGE_AGGREGATES[sorted_zone_keys]:
+            exchanges_to_fetch.append(domain_pair)
+    elif sorted_zone_keys in ENTSOE_EXCHANGE_DOMAIN_OVERRIDE:
+        exchanges_to_fetch.append(ENTSOE_EXCHANGE_DOMAIN_OVERRIDE[sorted_zone_keys])
     else:
-        domain1 = ENTSOE_DOMAIN_MAPPINGS[zone_key1]
-        domain2 = ENTSOE_DOMAIN_MAPPINGS[zone_key2]
-    # Grab exchange
-    # Import
-    try:
-        raw_exchange = query_function(domain1, domain2, session, target_datetime)
-    except Exception as e:
-        raise ParserException(
-            parser="ENTSOE.py",
-            message=f"Failed to query import for {zone_key1} -> {zone_key2}",
-            zone_key=sorted_zone_keys,
-        ) from e
-    if raw_exchange is not None:
-        imports = parse_exchange(
-            raw_exchange,
-            is_import=True,
-            sorted_zone_keys=sorted_zone_keys,
-            logger=logger,
-            is_forecast=forecast,
+        exchanges_to_fetch.append(
+            [ENTSOE_DOMAIN_MAPPINGS[zone_key1], ENTSOE_DOMAIN_MAPPINGS[zone_key2]]
         )
-        if imports:
-            # Export
-            try:
-                raw_exchange = query_function(
-                    domain2, domain1, session, target_datetime
-                )
-            except Exception as e:
-                raise ParserException(
-                    parser="ENTSOE.py",
-                    message=f"Failed to query export for {zone_key1} -> {zone_key2}",
-                    zone_key=sorted_zone_keys,
-                ) from e
-            if raw_exchange is not None:
-                exports = parse_exchange(
-                    xml_text=raw_exchange,
-                    is_import=False,
-                    sorted_zone_keys=sorted_zone_keys,
-                    logger=logger,
-                    is_forecast=forecast,
-                )
-                if imports and exports:
-                    exchanges = ExchangeList.merge_exchanges([imports, exports], logger)
-        return exchanges
-    else:
-        raise ParserException(
-            parser="entsoe.eu",
-            message=f"No exchange data found for {zone_key1} -> {zone_key2}",
+
+    def _fetch_and_parse_exchange(
+        domain_pair: list[str],
+        is_import: bool,
+    ) -> ExchangeList:
+        """
+        Internal function to fetch and parse exchange data
+        only used to avoid code duplication in the parent function.
+        """
+        domain1, domain2 = domain_pair if is_import else domain_pair[::-1]
+        try:
+            raw_exchange = query_function(domain1, domain2, session, target_datetime)
+        except Exception as e:
+            raise ParserException(
+                parser="ENTSOE.py",
+                message=f"Failed to query {'import' if is_import else 'export'} for {domain1} -> {domain2}",
+                zone_key=sorted_zone_keys,
+            ) from e
+        if raw_exchange is not None:
```

Comment:
A personal preference of mine would be to just raise if it's None and carry on with the rest.

Reply:
I'll change that when I add the new snapshot tests.